data-processing/common/upload_entities.py

```diff
@@ -309,3 +313,53 @@ def resolve_model(entity_identifier: EntityIdentifier) -> Optional[Entity]:
                     )
 
     return models
+
+
+def write_to_file(entity_infos: List[EntityUploadInfo], output_file_name: str) -> None:
+    # an attempt to make life easier if we change how we want to format this
+    FORMAT_VERSION = "v0"
+
+    logging.info(
+        "About to write %d entity infos to %s (version: %s).",
+        len(entity_infos),
+        output_file_name,
+        FORMAT_VERSION,
+    )
+    to_write = {
+        "version": FORMAT_VERSION,
+        "data": [dataclasses.asdict(entity_info) for entity_info in entity_infos],
+    }
+    already_exists_msg = (
+        f"File {output_file_name} already exists. "
+        + "Not overwriting. Entity infos will not be written."
+    )
+    assert not os.path.exists(output_file_name), already_exists_msg
+
+    with open(output_file_name, "w") as output_file:
+        json.dump(to_write, output_file)
+
+
+def save_entities(
+    s2_id: S2Id,
+    arxiv_id: ArxivId,
+    entity_infos: List[EntityUploadInfo],
+    data_version: Optional[int],
+    output_details: OutputDetails,
+    filename: str,
```

Comment:
How about having `output_path` that includes filename (and not just the dir) in `OutputDetails`? Then you don't need this param.

Reply:
Yeah I can see the appeal of that but I think it doesn't quite work...
Each upload command will have one instance of OutputDetails. However, a single upload command could write to more than one file. E.g. if you take a look at the `data-processing/entities/definitions/upload.py` file in the [branch](https://github.com/allenai/scholarphi/compare/chi-2021-demo...chloea-try-producing-json-take-1) that shows how this could be extended to the other scholarphi inner pipelines, the command using `upload_definitions()` writes to two files: `definitions_terms.jsonl` and `definitions_symbols.jsonl`.
(We could probably change that code to collect all the entities for that pipeline before uploading them to the db or writing them to a file, so that we could write them to a single file, but I think I'd rather have this extra argument, and keep it possible to write entities as we go).

Reply:
Oh I see. Nvm then.